include/alpaka/exec/ExecGpuCudaRt.hpp

```diff
@@ -282,22 +304,8 @@ namespace alpaka
                     auto const threadElemExtent(
                         workdiv::getWorkDiv<Thread, Elems>(task));
 
-                    dim3 gridDim(1u, 1u, 1u);
-                    dim3 blockDim(1u, 1u, 1u);
-                    // \FIXME: CUDA currently supports a maximum of 3 dimensions!
-                    for(auto i(static_cast<typename TDim::value_type>(0)); i<std::min(static_cast<typename TDim::value_type>(3), TDim::value); ++i)
-                    {
-                        reinterpret_cast<unsigned int *>(&gridDim)[i] = static_cast<unsigned int>(gridBlockExtent[TDim::value-1u-i]);
-                        reinterpret_cast<unsigned int *>(&blockDim)[i] = static_cast<unsigned int>(blockThreadExtent[TDim::value-1u-i]);
-
-                    }
-                    // Assert that all extent of the higher dimensions are 1!
-                    for(auto i(std::min(static_cast<typename TDim::value_type>(3), TDim::value)); i<TDim::value; ++i)
-                    {
-                        BOOST_VERIFY(gridBlockExtent[TDim::value-1u-i] == 1);
-                        BOOST_VERIFY(blockThreadExtent[TDim::value-1u-i] == 1);
-                        BOOST_VERIFY(threadElemExtent[TDim::value-1u-i] == 1);
```

Comment:
Could it be that `threadElemExtent` not checked any more because of the refactoring?

Reply:
You are right, thank you for reviewing this. I will fix it.

Reply:
It should be fixed now!